azure-pipelines.yml

```diff
@@ -194,3 +194,24 @@ jobs:
 
   - script: python setup.py sdist
     displayName: 'Build sdist'
+
+
+- job: DocsTest
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+  - bash: echo "##vso[task.prependpath]/usr/share/miniconda/bin"
+    displayName: Add conda to PATH
+  - script: conda env create -n test_env --quiet --file ci/doc.yml
+    displayName: Create Anaconda environment
+  - script: |
+      source activate test_env
+      # Replicate the exact environment created by the readthedocs CI
+      # conda install --yes --quiet -c pkgs/main mock pillow sphinx sphinx_rtd_theme
+      pip install -e .[all]
+    displayName: Replicate readthedocs CI environment
```

Comment:
install pvlib for doc build?

Reply:
I believe this is necessary to run the sphinx-gallery examples at least. I was inspired by xarray's Azure config and this is also [a step that xarray uses as well](https://github.com/pydata/xarray/blob/18e34cc8746ea2ce7be785edcaeee0bd7d4a0200/ci/azure/install.yml#L39). 

Based on the readthedocs.yml file, isn't this at least part of how ReadTheDocs builds our documentation for us? In our readthedocs.yml file it states that the `extra_requirements` are listed as `-all` and based on [The ReadTheDocs documentation](https://docs.readthedocs.io/en/latest/config-file/v1.html#python-extra-requirements) that is what is installed during their build of our docs. I'm inexperienced with ReadTheDocs so please let me know if I am missing something.

Reply:
I'm not sure we're on the same page here. My comment should have been a suggestion like

```
    displayName: Install pvlib[all] for doc build
```

Reply:
I like that better, thanks for the recommendation.